pkg/server/server.go

```diff
@@ -352,10 +408,14 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 			graph.WithLogger(s.logger),
 			graph.WithCacheTTL(s.checkQueryCacheTTL),
 		)
-		s.cachedCheckResolverCloser = cachedCheckResolver.Close
+		s.cachedCheckResolver = cachedCheckResolver
 
 		cachedCheckResolver.SetDelegate(localChecker)
-		cycleDetectionCheckResolver.SetDelegate(cachedCheckResolver)
+		if s.dispatchThrottlingCheckResolver != nil {
+			s.dispatchThrottlingCheckResolver.SetDelegate(cachedCheckResolver)
+		} else {
```

Comment:
I pulled your PR and annotated the code with comments showing how the circular linked list of resolvers is wired and it looks like this

<img width="1119" alt="image" src="https://github.com/openfga/openfga/assets/5374887/9a840333-1555-40e5-bf93-5eafc582825c">

The `if` branch looks incorrect.

Reply:
Please see https://github.com/openfga/openfga/blob/5810589f38dc5ba91b5a871e75050171a2927aef/pkg/server/server_test.go#L1532

Reply:
Never mind - it is correct, i forgot that in this `if`, the other `if` is also running